backend/src/v2/compiler/argocompiler/container.go

```diff
@@ -111,7 +133,7 @@ func (c *workflowCompiler) addContainerDriverTemplate() string {
 			},
 		},
 		Container: &k8score.Container{
-			Image:   c.driverImage,
+			Image:   GetDriverImage(),
```

Comment:
nit: It would be great if you can add some unit tests to this change. Not necessarily a blocker for merging this PR though.